### PR TITLE
SDK rewrite

### PR DESCRIPTION
**Pull request by 30PandaX**

### CODE REVIEW

**Review by k-fun — APPROVED** [review]

File: src/FunKit.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+        const wallet = new FunWallet(
+            {
+                users,
+                uniqueId
+            },
+            this.options,
+            await this.getChain(chainId)
+        )
+        await wallet.getAddress()
```
[the remark this answers is not in the document]
Comment (30PandaX) [reply src/FunKit.ts:91]: Yeah this line sounds more like a init than getAddress. More like a naming debt from before.

File: src/FunKit.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        return wallet
+    }
+
+    async getWallet(address: string, chainId?: number): Promise<FunWallet> {
+        const wallet = new FunWallet(address, this.options, await this.getChain(chainId))
+        wallet.getAddress()
```
[the remark this answers is not in the document]
Comment (30PandaX) [reply src/FunKit.ts:97]: Will fix!

File: src/FunKit.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    }
+
+    /**
+     * Get token balance of a wallet without create a fun wallet
+     **/
+    async getTokenBalance(tokenAddress: string, walletAddress: Address, chain?: Chain): Promise<string> {
+        return await Token.getBalance(tokenAddress, walletAddress, await this.getChain(chain), this.options.apiKey)
+    }
+
+    async getOps(opIds: Hex[], chainId?: string): Promise<any> {
```
[the remark this answers is not in the document]
Comment (30PandaX) [reply src/FunKit.ts:112]: Yep, thanks for catching!

File: src/FunKit.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (30PandaX) [reply src/FunKit.ts]: > Same comment, and also I think this should be `await`'ed

Will fix!

File: src/FunKit.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (30PandaX) [reply src/FunKit.ts:91]: > This is weird. We can keep it for now, but should have a comment like `Ensure wallet.address is populated before returning the wallet.`

Yeah this line sounds more like a init than getAddress. More like a naming debt from before.

**Pull request merged**